### PR TITLE
Fix Xpert styling issues, fix lingering bugs, and make improvements.

### PR DESCRIPTION
**Pull request by MichaelRoytman**
### Description

This pull request adds proper styling to the Xpert chatbot to closely resemble the Hackathon UI/UIX and also introduces a few fixes to the Xpert chatbot.

These changes include:

**Styling**
* converting Tailwind classes to Bootstrap classes, when equivalent Bootstrap classes exist
* convert Tailwind classes to CSS, when equivalent Bootstrap classes do not exist
* convert CSS to SCSS to improve readability
* leverage flexbox to remove absolute positioning, where possible
* show and hide the Sidebar by conditionally rendering the component instead of using opacity and positioning

**Behavior**
* correctly dispatch the `addChatMessage` thunk action creator
* move the API loading state to Redux
* move API call from `Xpert` widget to Sidebar component
* store serializable date value in Redux
* fix propType for `chatbotContainerRef`
* scroll down to bottom of the chat window after reopening the Xpert chatbot

JIRA: [MST-2034](https://2u-internal.atlassian.net/browse/MST-2034) (private)

### Demos

**Before**

https://github.com/edx/frontend-lib-learning-assistant/assets/11871801/31f8452b-bce5-4a6a-8dcd-fb8e812f2395

**After**

Please note that the Xpert responses are mocked out. The Xpert chatbot is not making actual API calls in this demo.

https://github.com/edx/frontend-lib-learning-assistant/assets/11871801/2faa34a6-00b7-4e61-ba35-ee7cc2ed79cb

### CODE REVIEW

File: src/components/ChatBox/index.jsx [hunk 1]
```diff
@@ -1,27 +1,20 @@
...
 import PropTypes from 'prop-types';
 import Message from '../Message';
-import './ChatBox.css';
+import './ChatBox.scss';
 
 // container for all of the messages
-const ChatBox = ({ messageList, chatboxContainerRef }) => {
-  const [isResponseLoading, setResponseLoading] = useState(false);
-
-  useEffect(() => {
```
Comment (MichaelRoytman) [inline src/components/ChatBox/index.jsx:10]: I've moved the API loading state into Redux. Before, it was part of the Chatbox component state. We were deriving the loading state from state in the Redux store that was being modified by a few different action creators, which meant that the "Xpert is thinking..." message wasn't showing properly.

Moving this state into the Redux store fixes the display of the "Xpert is thinking..." message, makes it easier to define and set this value, and makes it available to other components in the tree that may need it later (e.g. disabling the button during loading).

File: src/components/ChatBox/index.jsx [hunk 2]
```diff
@@ -31,9 +24,9 @@ ChatBox.propTypes = {
   messageList: PropTypes.arrayOf(PropTypes.shape({
     role: PropTypes.string.isRequired,
     content: PropTypes.string.isRequired,
-    timestamp: PropTypes.instanceOf(Date).isRequired,
+    timestamp: PropTypes.string.isRequired,
   })).isRequired,
-  chatboxContainerRef: PropTypes.string.isRequired,
+  chatboxContainerRef: PropTypes.object.isRequired,
```
Comment (MichaelRoytman) [inline src/components/ChatBox/index.jsx:29]: The `chatboxContainerRef` is an object, not a string, so this was failing.

File: src/components/Sidebar/index.jsx [hunk 3]
```diff
@@ -23,32 +25,35 @@ const Sidebar = ({
...
+            requestAnimationFrame(scroll);
+          }
+        };
 
-      requestAnimationFrame(scroll);
+        requestAnimationFrame(scroll);
+      }
     }
-  }, [messageList]);
+  }, [messageList, isOpen]);
```
Comment (MichaelRoytman) [inline src/components/Sidebar/index.jsx:56]: Consequently, when we close and reopen the Sidebar, the component is remounted, and we lose the scroll position. Adding `isOpen` to the dependency array causes the Sidebar to scroll to the proper position. The learner does see this after the Sidebar opens, which may not be ideal. Please see the demo video.

File: src/components/Sidebar/index.jsx [hunk 4]
```diff
@@ -62,6 +67,7 @@ const Sidebar = ({
     event.preventDefault();
     if (currentMessage) {
       dispatch(addChatMessage('user', currentMessage));
+      dispatch(getChatResponse(courseId));
```
Comment (MichaelRoytman) [inline src/components/Sidebar/index.jsx:70]: I've moved the API call logic from the Xpert widget to the Sidebar component. Currently, the API call to the Xpert backend is occurring as as a side effect of changes to the messageList Redux state in this useEffect, which is the parent component of the Sidebar. This is a side-effect that is difficult to follow. After this change, the API call occurs directly as a consequence of the user's action - enter or button click to submit the message. This makes it easier to follow how an API call is made.

File: src/data/thunks.js [hunk 5]
```diff
@@ -3,18 +3,24 @@ import fetchChatResponse from './api';
...
+  setApiIsLoading,
 } from './slice';
 
 export function addChatMessage(role, content) {
   return (dispatch, getState) => {
     const { messageList, conversationId } = getState().learningAssistant;
+
+    // Redux recommends only serializable values in the store, so we'll stringify the timestap to store in Redux.
+    // When we need to operate on the Date object, we'll deserialize the string.
+    const timestamp = new Date();
```
Comment (alangsto) [inline src/data/thunks.js:18]: Nice update 👍

File: src/components/Sidebar/index.jsx (hunk 4 above)
Comment (alangsto) [inline src/components/Sidebar/index.jsx:70]: I like that this is moved closer to the action that triggers it

**Review by alangsto — APPROVED** [review]
Incredible work! I don't see anything that jumps out right away, but I'll take a more detailed look at the CSS too

**Pull request merged**